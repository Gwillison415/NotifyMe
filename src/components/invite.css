/* USER PROFILE PAGE */
.useravatar{
  padding-right: 50em;
  padding-bottom: 25px;
  margin-bottom: 15px;
}
 .card {
    margin-top: 10px;
    padding: 25px;
    background-color: rgba(214, 224, 226, 0.2);
    /*-webkit-border-top-left-radius:5px;
    -moz-border-top-left-radius:5px;*/
    border-top-left-radius:5px;
    -webkit-border-top-right-radius:5px;
    -moz-border-top-right-radius:5px;
    border-top-right-radius:5px;
    -webkit-box-sizing: border-box;
    -moz-box-sizing: border-box;
    box-sizing: border-box;
}

.card.hovercard {
    position: relative;
    padding-top: 0;
    overflow: hidden;
    text-align: center;
    background-color: #fff;
    background-color: rgba(255, 255, 255, 1);
}
.card.hovercard .card-background {
    height: 130px;
}
.card-background img {
    -webkit-filter: blur(25px);
    -moz-filter: blur(25px);
    -o-filter: blur(25px);
    -ms-filter: blur(25px);
    filter: blur(3px);
    margin-left: -100px;
    margin-top: -200px;
    min-width: 125%;
}
.card.hovercard .useravatar {
    position: absolute;

    top: 15px;
    left: 0;
    right: 0;
}
.card.hovercard .useravatar img {
    width: 100px;
    height: 100px;
    max-width: 100px;
    max-height: 100px;
    -webkit-border-radius: 50%;
    -moz-border-radius: 50%;
    border-radius: 50%;
    border: 5px solid rgba(255, 255, 255, 0.5);
}
.card.hovercard .card-body {
    position: absolute;
    bottom: 14px;
    left: 0;
    right: 0;
}
.card.hovercard .card-body .card-title {
    padding:0 5px;
    font-size: 20px;
    line-height: 1;
    color: #8F778C;
    background-color: rgba(255, 255, 255, 0.1);
    -webkit-border-radius: 4px;
    -moz-border-radius: 4px;
    border-radius: 4px;
    padding-right: 45em;
}

.card.hovercard .card-body {
    overflow: hidden;
    font-size: 12px;
    line-height: 20px;
    color: #737373;
    text-overflow: ellipsis;
}
.card.hovercard .bottom {
    padding: 0 20px;
    margin-bottom: 1px;
}
.btn-pref .btn {
    -webkit-border-radius:0 !important;
}
.card-body {
  margin-bottom: -15px;
}

.extraClass {
 font-size: 20px !important;
 pointer-events: auto !important;
 &:hover {
  visibility: visible !important;
  opacity: 1 !important;
 }
}
