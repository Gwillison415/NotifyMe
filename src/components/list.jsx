import {findDOMNode} from 'react-dom';
import React from 'react';
import $ from 'jquery';

class FullDesc extends React.Component {
 constructor() {
 super();
 }
handleToggle = () => {
 const el = findDOMNode(this.refs.toggle);
 $(el).slideToggle();
 };
render() {
 return (
 <div className="list-group">
  <ul className="profile-info">
   <li>
     <span className="info-title">User Name : </span> Shuvo Habib
   </li>
 </ul>
<ul className="profile-info additional-profile-info-list" ref="toggle">
  <li>
    <span className="info-email">Office Email</span> me@shuvohabib.com
  </li>
 </ul>
  <div className="ellipsis-click" onClick={this.handleToggle}>
    <i className="fa-ellipsis-h"/>
  </div>
 </div>
 );
 }
}
export default FullDesc;

//
// <div class="list-group">
// <a href="#" class="list-group-item pull-right list-group-item-action active">
// Cras justo odio
// </a>
// <a href="#" class="list-group-item list-group-item-action">Dapibus ac facilisis in</a>
// <a href="#" class="list-group-item list-group-item-action">Morbi leo risus</a>
// <a href="#" class="list-group-item list-group-item-action">Porta ac consectetur ac</a>
// <a href="#" class="list-group-item list-group-item-action disabled">Vestibulum at eros</a>
// </div>
