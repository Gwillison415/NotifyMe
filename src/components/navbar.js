import React from 'react';
import {connect} from 'react-redux'
import {bindActionCreators} from 'redux'
import { Collapse, Navbar, NavbarToggler, NavbarBrand, Nav, NavItem, NavLink } from 'reactstrap';
import { UncontrolledDropdown, DropdownToggle, DropdownMenu, DropdownItem } from 'reactstrap';
import {handleUpdates} from '../actions';
 class NavToolBar extends React.Component {

  render() {
    return (
      <div>
        <Navbar color="blue" light expand="md">
          <NavbarBrand href="/">NotifyMe</NavbarBrand>
          <UncontrolledDropdown size="small">
            <DropdownToggle caret id="branchInput">

            </DropdownToggle>
            <DropdownMenu >
              <DropdownItem> Time since invitation:</DropdownItem>
              {/* <DropdownItem onClick={()=>{this.props.handleUpdates()}}> Trigger Updated invitations</DropdownItem>
              <DropdownItem divider/> */}
              <DropdownItem > <a href="https://github.com/Gwillison415/NotifyMe">Github Repo</a></DropdownItem>
            </DropdownMenu>
          </UncontrolledDropdown>


        </Navbar>
      </div>
    );
  }
}

const mapStateToProps = state => ({invites: state.invites})
const mapDispatchToProps = dispatch => bindActionCreators({
  handleUpdates
}, dispatch)

export default connect(mapStateToProps, mapDispatchToProps)(NavToolBar);
