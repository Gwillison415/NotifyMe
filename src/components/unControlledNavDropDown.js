import React from 'react';
import { connect } from 'react-redux'
import { bindActionCreators } from 'redux'
import { Container, UncontrolledDropdown, DropdownToggle, DropdownMenu, DropdownItem } from 'reactstrap';

export const UCDropdown = ({
    inviteURL,
    elapsedTimeHumanReadable,
}) => {

    return (

        <UncontrolledDropdown size="small">
          <DropdownToggle caret id="branchInput">
            More Info
          </DropdownToggle>
          <DropdownMenu >
            <DropdownItem> Time Open:</DropdownItem>
            <DropdownItem>{elapsedTimeHumanReadable}</DropdownItem>
            <DropdownItem divider/>
            <DropdownItem > <a href={inviteURL}>{inviteURL}</a></DropdownItem>
          </DropdownMenu>
        </UncontrolledDropdown>

    );

}




const mapDispatchToProps = dispatch => bindActionCreators({

}, dispatch)

export default connect(
  null,
  mapDispatchToProps
)(UCDropdown);
