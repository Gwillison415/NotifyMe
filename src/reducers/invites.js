import {
  INVITES_REQUEST_STARTED,
  INVITES_REQUEST_SUCCESS,
  UPDATE_JSON,
  TOGGLE_JOIN,
} from '../actions'

const initialState = {
  ids: [],
  invites: [],
  invitesById: {},
  fetchingInvites: true,

}
let urlRegex = new RegExp(/https?:\/\/(?:www\.|(?!www))[a-zA-Z0-9][a-zA-Z0-9-]+[a-zA-Z0-9]\.[^\s]{2,}|www\.[a-zA-Z0-9][a-zA-Z0-9-]+[a-zA-Z0-9]\.[^\s]{2,}|https?:\/\/(?:www\.|(?!www))[a-zA-Z0-9]\.[^\s]{2,}|www\.[a-zA-Z0-9]\.[^\s]{2,}/)
let joinRequest = new RegExp(/\bjoin Situation\b/)
let filterDupesObj = {};
let statsObj = {
  read: 0,
  unread: 0,
};

function createState(json, incomingState, isUpdate = false ) {


  const state = incomingState



    let count = 0

  json.forEach((invite) => {
    //parse the invite message for truly relevant info given limited 'real estate'
    let subject = invite.invite.match(/\[[^\]]+\]/, 'g')
    //pull string out of RegExp OBJ
    // subject = subject[0].slice(1, subject[0].length - 1)
    let url = invite.invite.match(urlRegex, 'mg')

    //bool to indicate when to put up a 'join' button
    let isJoinRequest = joinRequest.test(invite.invite)

    if (state.invitesById.hasOwnProperty(invite.sig_id)) {

      if (state.invitesById[invite.sig_id].subject === subject[0].slice(1, subject[0].length - 1)) {
        console.log('duplicate entry not added');
        return;
      } else {
        // write new entry into state tree
        state.invites.push(Object.assign(invite, {subject: subject[0].slice(1, subject[0].length - 1), url: url[0], isJoinRequest}))

        state.ids = state.ids.concat(invite.invite_id);
        state.invitesById[invite.sig_id] = {};
        state.invitesById[invite.sig_id].invite = invite.invite;
        state.invitesById[invite.sig_id].sender_id = invite.sender_id;
        state.invitesById[invite.sig_id].subject = invite.subject;
        state.invitesById[invite.sig_id].url = invite.url;
        state.invitesById[invite.sig_id].invite_id = invite.invite_id;
        state.invitesById[invite.sig_id].isJoinRequest = invite.isJoinRequest;
        state.invitesById[invite.sig_id].status = invite.status;
        state.invitesById[invite.sig_id].vector = invite.vector;
        state.invitesById[invite.sig_id].sig_id = invite.sig_id;
        state.invitesById[invite.sig_id].percentComplete = 0;
      }


    } else {
      // If there isn't an entry, write entry into state tree
      state.invites.push(Object.assign(invite, {subject: subject[0].slice(1, subject[0].length - 1), url: url[0], isJoinRequest}))

      state.ids = state.ids.concat(invite.invite_id);
      state.invitesById[invite.sig_id] = {};
      state.invitesById[invite.sig_id].invite = invite.invite;
      state.invitesById[invite.sig_id].sender_id = invite.sender_id;
      state.invitesById[invite.sig_id].subject = invite.subject;
      state.invitesById[invite.sig_id].url = invite.url;
      state.invitesById[invite.sig_id].invite_id = invite.invite_id;
      state.invitesById[invite.sig_id].isJoinRequest = invite.isJoinRequest;
      state.invitesById[invite.sig_id].status = invite.status;
      state.invitesById[invite.sig_id].vector = invite.vector;
      state.invitesById[invite.sig_id].sig_id = invite.sig_id;
      state.invitesById[invite.sig_id].percentComplete = 0;
    }

  });
  return { ...state };
}


export default (state = initialState, action) => {

  switch (action.type) {
    case INVITES_REQUEST_STARTED:
      return createState(action.response.invites, state);
    case INVITES_REQUEST_SUCCESS:
      return {
        ...state,
        fetchingInvites: false,
        // invites: action.invites,
      }
    case TOGGLE_JOIN:
    return {
      ...state,
      invites: toggleProperty(state.invites, action.invite, "someprop")
    }
    case UPDATE_JSON:
      return createState(action.response.invites, state, action.isUpdate);
    default:
      return state
  }
}


function toggleProperty(invites, invite, property) {
  console.log(property);
  const index = invites.indexOf(invite)
  return [
    ...invites.slice(0, index),
    { ...invite, [property]: !invite[property] },
    ...invites.slice(index + 1),
  ];
}
