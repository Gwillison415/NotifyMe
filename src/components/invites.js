import React from 'react'
import {InviteComponent} from './invite';
import {connect} from 'react-redux'


import {
  Card,
  CardDeck,
  Container,
  Col,
  Row
} from 'reactstrap';


// class InvitesComponent extends Component {
//   constructor() {
//     // this.props.invites
//   }
// }
export const InvitesComponent = ({invites}) => {
  // console.log('state in Component', this.props);
  let invitations = [ [], [] ]
  invites.forEach((invite, idx) => {
  

      if (invite.status === "read") {
        invitations[0].push(<Card key={idx}>
          <InviteComponent inviteID={invite.invite_id} sender={invite.sender_id} inviteMsg={invite.invite}  vector={invite.vector} subject={invite.subject} status={invite.status} sig_id={invite.sig_id} inviteSelected={invite.selected} inviteTime={invite.invite_time} invite={invite} />
        </Card>)
      } else {
        invitations[1].push(<Card key={idx}>
          <InviteComponent inviteID={invite.invite_id} sender={invite.sender_id} subject={invite.subject} inviteMsg={invite.invite}  vector={invite.vector} status={invite.status} sig_id={invite.sig_id} inviteSelected={invite.selected} inviteTime={invite.invite_time} invite={invite} />
        </Card>)
      };

  })

// isJoinRequest={isJoinRequest} inviteSubject={subject[0]} inviteURL={url[0]}

    return (
      <Container>
        <CardDeck width="100%">
          <Row>
            <Col  sm="6" xs={{ size: 12,  pull: 1} }>
              <h2 >Unread Messages</h2>
              {invitations[1]}
            </Col>
            <Col sm="6" xs={{ size: 12,  pull: 1}  }>
              <h2>Read Messages</h2>
              {invitations[0]}
            </Col>
          </Row>
        </CardDeck>
      </Container>
    )

}

export const mapStateToProps = (state, ownProps) => {

  const invitesById = state.invites.invitesById;
  const invites = state.invites.invites


  return {
    invites,
    invitesById,
  };
};

// export const mapDispatchToProps = dispatch =>
//   bindActionCreators(
//     {
//       retrieveMilestones,
//     },
//     dispatch,
//   );

export default connect(mapStateToProps, null)(InvitesComponent);
// export default InvitesComponent
