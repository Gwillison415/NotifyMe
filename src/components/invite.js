import React from 'react'
import {connect} from 'react-redux';
import {UCDropdown} from './unControlledNavDropDown';
import {toggleProperty} from '../actions';
import {bindActionCreators} from 'redux'

import {
  Card,
  Button,
  CardTitle,
  CardText,
  CardImg,
  CardBody,
  CardSubtitle
} from 'reactstrap';

// import {imageContent} from '../static/imageSources';

// import {toggle} from '../actions';
import NOTIFY_API from '../utils/Api';
import './invite.css';

var moment = require('moment');

export const createVectorCSSClass = (source) => {
  switch (source) {
    case "Twitter":
      return "fa fa-twitter"
    case "Internal":
      return "fa fa-home"
    case "Email":
      return "fa fa-envelope-o"
    default:
      return "fa fa-chain-broken"
  }
}

export const InviteComponent = ({
  sender,
  inviteKey,
  inviteMsg,
  inviteSubject,
  inviteURL,
  vector,
  status,
  situationID,
  inviteTime,
  subject,
  invite,
  isJoinRequest,
  sig_id

}) => {
  let sourceClass = createVectorCSSClass(vector);
  let timeCreatedHumanReadable = NOTIFY_API.returnDate(inviteTime);
  let elapsedTimeHumanReadable = NOTIFY_API.getDateTimeSince(new Date(moment.unix(inviteTime)))
  let statusColor = status === "read"
    ? "secondary"
    : "danger"

  return (<Card color={statusColor}>
    <CardImg className="top" width="110%" src={NOTIFY_API.findAvatar(sender)} alt="Card image cap"/>
    <CardBody>
      <CardTitle width="140">{subject}</CardTitle>
      <CardSubtitle className={`${sourceClass}`}>
        <div>
          Source:  {vector}
        </div>
        <div>
          sig_id:{sig_id}
        </div>
      </CardSubtitle>
      <CardText>
        {`Created On: \n ${timeCreatedHumanReadable}`}

      </CardText>
      {
        isJoinRequest
          ? <div>
            <span>
                <Button href={inviteURL} color="primary">{status.toUpperCase()}
                  </Button>

              </span>
              <span>
                <Button onClick={() => {toggleProperty(invite, "isJoinRequest")}} >Join Now</Button>
              </span>
          </div>
          : null
      }
      <UCDropdown size="auto" elapsedTimeHumanReadable={elapsedTimeHumanReadable} inviteURL={inviteURL} color="success"></UCDropdown>

    </CardBody>
  </Card>)
}

export const mapStateToProps = (state, ownProps) => {
  const id = ownProps.inviteID;
  const subject = state.invites.invitesById[id].subject;
  const sender = state.invites.invitesById[id].sender_id;
  const isJoinRequest = state.invites.invitesById[id].isJoinRequest;
  return {subject, isJoinRequest, sender};
};

const mapDispatchToProps = dispatch => bindActionCreators({
  toggleProperty
}, dispatch)

export default connect(mapStateToProps, mapDispatchToProps)(InviteComponent);
// export default InviteComponent;
