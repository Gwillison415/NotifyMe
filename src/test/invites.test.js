import React from 'react';
import toJson from 'enzyme-to-json';
import {shallow, configure} from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

import {InvitesComponent, mapStateToProps, mapDispatchToProps} from '../components/invites';
import invitationsTest from '../static/invitationsTest';
configure({adapter: new Adapter()});
const state = {
  invites: {
    ids: [
      25121, 23951, 92394, 49120, 10293
    ],
    invites: invitationsTest,
    invitesById: {

      10293: {
        invite: "The Owner has invited you to join Situation 10293 [Mail intermittendly available] Please follow this link to open Situation Room: http://www.moogsoft.com",
        sender_id: "mike",
        subject: "Mail intermittendly available",
        url: "http://www.moogsoft.com",
        invite_id: 5,
        isJoinRequest: true,
        status: "read",
        vector: "Email",
        sig_id: 10293,
        invite_time: 1399292261
      },

      23951: {
        invite: "The Owner has invited you to join Situation 23951 [Slow access to Purchasing system] Please follow this link to open Situation Room: http://www.moogsoft.com",
        sender_id: "andrew",
        subject: "Slow access to Purchasing system",
        url: "http://www.moogsoft.com",
        invite_id: 2,
        isJoinRequest: true,
        status: "read",
        vector: "Email",
        sig_id: 23951,
        invite_time: 1398992261
      },
      25121: {
        invite: "The Owner has invited you to join Situation 25121 [Mail system down] Please follow this link to open Situation Room: http://www.moogsoft.com",
        sender_id: "jeff",
        subject: "Mail system down",
        url: "http://www.moogsoft.com",
        invite_id: 1,
        isJoinRequest: true,
        status: "read",
        vector: "Internal",
        sig_id: 25121,
        invite_time: 1398892261
      },
      49120: {
        invite: "The Owner has invited you to join Situation 49120 [No access to AWS account] Please follow this link to open Situation Room: http://www.moogsoft.com",
        sender_id: "adrian",
        subject: "No access to AWS account",
        url: "http://www.moogsoft.com",
        invite_id: 4,
        isJoinRequest: true,
        status: "unread",
        vector: "Internal",
        sig_id: 49120,
        invite_time: 1399192261
      },
      92394: {
        invite: "The Owner has invited you to join Situation 92394 [End-users reporting no service in London] Please follow this link to open Situation Room: http://www.moogsoft.com",
        sender_id: "neil",
        subject: "End-users reporting no service in London",
        url: "http://www.moogsoft.com",
        invite_id: 3,
        isJoinRequest: true,
        status: "read",
        vector: "Twitter",
        sig_id: 92394,
        invite_time: 1399092261
      }
    },
    statsObj: {}
  }

}

describe('Invites component', () => {
  it('should render a component with props as specified ', () => {
    const retrieveInvites = jest.fn();
    const component = shallow(<InvitesComponent invites={state.invites.invites} invitesById={state.invites.invitesById} ids={state.invites.ids}/>);
    expect(toJson(component)).toMatchSnapshot();
  });

  it("map's given State To Props ", () => {
    const expected = {
      ids: state.invites.ids,
      invitesById:state.invites.invitesById,
    };
    expect(mapStateToProps(state)).toEqual(expected);
  });

  it('maps component dispatches to props', () => {
    const dispatch = jest.fn();
    expect(mapDispatchToProps(dispatch)).toHaveProperty('clearData');
  });

  it('Should be a Card Deck', () => {
    const component = shallow(<InvitesComponent invites={state.invites.invites} invitesById={state.invites.invitesById} ids={state.invites.ids}/>);
    // console.log('log??????', component.find([width="100%"]));
    expect(component.find('div.container').exists()).toBe(true);
  });
});
