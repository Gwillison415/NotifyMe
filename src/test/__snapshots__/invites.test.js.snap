// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Invites component should render a component with props as specified  1`] = `
<Container
  tag="div"
>
  <CardDeck
    tag="div"
    width="100%"
  >
    <Row
      tag="div"
    >
      <Col
        sm="6"
        tag="div"
        widths={
          Array [
            "xs",
            "sm",
            "md",
            "lg",
            "xl",
          ]
        }
        xs={
          Object {
            "pull": 1,
            "size": 12,
          }
        }
      >
        <h2
          className="text-warning"
        >
          Unread Messages
        </h2>
        <Card
          key="3"
          tag="div"
        >
          <Component
            invite={
              Object {
                "invite": "The Owner has invited you to join Situation 49120 [No access to AWS account] Please follow this link to open Situation Room: http://www.moogsoft.com",
                "invite_id": 4,
                "invite_time": 1399192261,
                "isJoinRequest": true,
                "sender_id": "adrian",
                "sig_id": 49120,
                "status": "unread",
                "subject": "No access to AWS account",
                "url": "http://www.moogsoft.com",
                "vector": "Internal",
              }
            }
            inviteID={4}
            inviteMsg="The Owner has invited you to join Situation 49120 [No access to AWS account] Please follow this link to open Situation Room: http://www.moogsoft.com"
            inviteTime={1399192261}
            inviteURL="http://www.moogsoft.com"
            isJoinRequest={true}
            sender="adrian"
            sig_id={49120}
            status="unread"
            subject="No access to AWS account"
            vector="Internal"
          />
        </Card>
      </Col>
      <Col
        sm="6"
        tag="div"
        widths={
          Array [
            "xs",
            "sm",
            "md",
            "lg",
            "xl",
          ]
        }
        xs={
          Object {
            "pull": 1,
            "size": 12,
          }
        }
      >
        <h2
          className="text-info"
        >
          Read Messages
        </h2>
        <Card
          key="0"
          tag="div"
        >
          <Component
            invite={
              Object {
                "invite": "The Owner has invited you to join Situation 25121 [Mail system down] Please follow this link to open Situation Room: http://www.moogsoft.com",
                "invite_id": 1,
                "invite_time": 1398892261,
                "isJoinRequest": true,
                "sender_id": "jeff",
                "sig_id": 25121,
                "status": "read",
                "subject": "Mail system down",
                "url": "http://www.moogsoft.com",
                "vector": "Internal",
              }
            }
            inviteID={1}
            inviteMsg="The Owner has invited you to join Situation 25121 [Mail system down] Please follow this link to open Situation Room: http://www.moogsoft.com"
            inviteTime={1398892261}
            inviteURL="http://www.moogsoft.com"
            isJoinRequest={true}
            sender="jeff"
            sig_id={25121}
            status="read"
            subject="Mail system down"
            vector="Internal"
          />
        </Card>
        <Card
          key="1"
          tag="div"
        >
          <Component
            invite={
              Object {
                "invite": "The Owner has invited you to join Situation 23951 [Slow access to Purchasing system] Please follow this link to open Situation Room: http://www.moogsoft.com",
                "invite_id": 2,
                "invite_time": 1398992261,
                "isJoinRequest": true,
                "sender_id": "andrew",
                "sig_id": 23951,
                "status": "read",
                "subject": "Slow access to Purchasing system",
                "url": "http://www.moogsoft.com",
                "vector": "Email",
              }
            }
            inviteID={2}
            inviteMsg="The Owner has invited you to join Situation 23951 [Slow access to Purchasing system] Please follow this link to open Situation Room: http://www.moogsoft.com"
            inviteTime={1398992261}
            inviteURL="http://www.moogsoft.com"
            isJoinRequest={true}
            sender="andrew"
            sig_id={23951}
            status="read"
            subject="Slow access to Purchasing system"
            vector="Email"
          />
        </Card>
        <Card
          key="2"
          tag="div"
        >
          <Component
            invite={
              Object {
                "invite": "The Owner has invited you to join Situation 92394 [End-users reporting no service in London] Please follow this link to open Situation Room: http://www.moogsoft.com",
                "invite_id": 3,
                "invite_time": 1399092261,
                "isJoinRequest": true,
                "sender_id": "neil",
                "sig_id": 92394,
                "status": "read",
                "subject": "End-users reporting no service in London",
                "url": "http://www.moogsoft.com",
                "vector": "Twitter",
              }
            }
            inviteID={3}
            inviteMsg="The Owner has invited you to join Situation 92394 [End-users reporting no service in London] Please follow this link to open Situation Room: http://www.moogsoft.com"
            inviteTime={1399092261}
            inviteURL="http://www.moogsoft.com"
            isJoinRequest={true}
            sender="neil"
            sig_id={92394}
            status="read"
            subject="End-users reporting no service in London"
            vector="Twitter"
          />
        </Card>
        <Card
          key="4"
          tag="div"
        >
          <Component
            invite={
              Object {
                "invite": "The Owner has invited you to join Situation 10293 [Mail intermittendly available] Please follow this link to open Situation Room: http://www.moogsoft.com",
                "invite_id": 5,
                "invite_time": 1399292261,
                "isJoinRequest": true,
                "sender_id": "mike",
                "sig_id": 10293,
                "status": "read",
                "subject": "Mail intermittendly available",
                "url": "http://www.moogsoft.com",
                "vector": "Email",
              }
            }
            inviteID={5}
            inviteMsg="The Owner has invited you to join Situation 10293 [Mail intermittendly available] Please follow this link to open Situation Room: http://www.moogsoft.com"
            inviteTime={1399292261}
            inviteURL="http://www.moogsoft.com"
            isJoinRequest={true}
            sender="mike"
            sig_id={10293}
            status="read"
            subject="Mail intermittendly available"
            vector="Email"
          />
        </Card>
      </Col>
    </Row>
  </CardDeck>
</Container>
`;
